docs/source/basics/tokenizers.rst

```diff
@@ -0,0 +1,239 @@
+.. _tokenizers_usage_label:
+
+==========
+Tokenizers
+==========
+
+Tokenizers are a key component of any LLM. They convert raw text into token IDs, which index into embedding vectors that are
+understood by the model.
+
+In torchtune, tokenizers play the role of converting Messages object into token IDs and any necessary model-specific special tokens.
+They use byte-pair encoding algorithms to convert raw strings into integer IDs. These are usually based on SentencePiece or TikToken.
+
+Downloading tokenizers from Hugging Face
+----------------------------------------
+
+Models hosted on Hugging Face are also distributed with the tokenizers they were trained with. These are automatically downloaded alongside
+model weights when using ``tune download``. For exmple, this command downloads the Mistral-7B model weights and tokenizer:
+
+.. code-block:: bash
+
+    tune download mistralai/Mistral-7B-v0.1 --output-dir /tmp/Mistral-7B-v0.1 --hf-token <HF_TOKEN>
+    cd /tmp/Mistral-7B-v0.1/
+    ls tokenizer.model
+    # tokenizer.model
```

Comment:
What does this mean?

Reply:
Separately I find the usage of `#` for ipython output confusing. Can we use something else? (Shouldn't be the same thing we use to denote code comments)

Reply:
>>> to indicate REPL?

Reply:
We've been using # everywhere, so I will stick with this and we can refactor it across the docs later